blockchain/beacon/src/drand.rs

```diff
@@ -34,6 +35,40 @@ impl DrandPublic {
     }
 }
 
+pub struct Schedule<T>(pub Vec<BeaconPoint<T>>)
+where
+    T: Beacon;
+
+impl<T> Schedule<T>
+where
+    T: Beacon,
+{
+    pub fn beacon_for_epoch(&self, e: ChainEpoch) -> Result<&T, Box<dyn error::Error>> {
+        if let Some(beacon_point) = self
+            .0
+            .iter()
+            .rev()
+            .find(|beacon| beacon.start == e)
+            .map(|s| &s.beacon)
+        {
+            Ok(&*beacon_point)
+        } else {
+            self.0
+                .first()
+                .map(|s| &*s.beacon)
+                .ok_or_else(|| Box::from("Could not get first_value of beacon in beacon_for_epoch"))
+        }
+    }
```

Comment:
Should maybe use shared logic as in `types::version` or connect it in some way so we don't have two similar functionalities not connected?

Reply:
I dont see how this is connected to anything in types::version...

Reply:
ah my bad, was a misread on my part